src/Umbraco.Core/Persistence/Repositories/Implement/ContentTypeRepository.cs

```diff
@@ -287,6 +287,16 @@ protected override void PersistUpdatedItem(IContentType entity)
                 entity.SortOrder = maxSortOrder + 1;
             }
 
+            // Update property variations based on the content type variation
+            foreach (var propertyType in entity.PropertyTypes)
+            {
+                // Determine variation for the property type.
+                // The property is only considered culture variant when the base content type is also culture variant.
+                // The property is only considered segment variant when the base content type is also segment variant.
+                // Example: Culture variant content type with a Culture+Segment variant property type will become ContentVariation.Culture
+                propertyType.Variations = entity.Variations & propertyType.Variations;
+            }
+
```

Comment:
This code auto corrects property type variations based on their content type variations. It was originally done *after* calling the validation method which we corrected [here](https://github.com/umbraco/Umbraco-CMS/pull/6208/files#diff-d367ba2cc01d9cb6029082d5bae70f1fR511-R517). However, unit tests were now failing. Apparently there is a hidden "feature" that property type variations were auto corrected to a valid variation based on their content type. The auto correction was done after the validation (??) which caused the unit tests to now fail. In order to fix the unit tests we had to perform the auto correct before the validation. 

It might seem like it's a big chance but the functionality should be preserved and now the validation method now covers all inputs.

Reply:
Interesting. I suppose however that this auto-correct code needs to exist on the `ContentTypeRepositoryBase` since one day we will probably have media types allowing variance as well. If we are auto-correcting, is there any point in validating? This would seem to always be true in that case. 

It turns out I added that validation check as part of this PR https://github.com/umbraco/Umbraco-CMS/pull/6019 and in this commit: https://github.com/umbraco/Umbraco-CMS/commit/9538981730d71698e90a00aba92fea700b587eef

... but i honestly don't recall all of the details of why but it was probably more of a safety check since this was quite a difficult bug to fix. It seems to me though, if you are changing the variance of a doc type, the of course all property types will need to be auto-adjusted and there doesn't seem to be a need for validation?